src/TLuaInterpreter.cpp

```diff
@@ -11728,35 +11728,264 @@ int TLuaInterpreter::sendSocket( lua_State * L )
+    int hport;
+    if (mudlet::self()->mpIrcClientMap[pHost]) {
+        hname = mudlet::self()->mpIrcClientMap[pHost]->getHostName();
+        hport = mudlet::self()->mpIrcClientMap[pHost]->getHostPort();
+    } else {
+        hname = dlgIRC::readIrcHostName(pHost);
+        hport = dlgIRC::readIrcHostPort(pHost);
+    }
+
+    lua_pushstring(L, hname.toUtf8().constData());
+    lua_pushinteger(L, hport);
+    return 2;
+}
+
+/** ircGetChannels();
+ *  Returns a string containing channels separated by a space.
+ *  If a client is active the list contains channels currently joined.
+ *  Otherwise the list is read from IRC client settings.
+ */
+int TLuaInterpreter::ircGetChannels(lua_State* L)
+{
+    Host* pHost = &getHostFromLua(L);
+    QStringList channels;
+    if (mudlet::self()->mpIrcClientMap[pHost]) {
+        channels = mudlet::self()->mpIrcClientMap[pHost]->getChannels();
+    } else {
+        channels = dlgIRC::readIrcChannels(pHost);
+    }
+
+    lua_pushstring(L, channels.join(" ").toUtf8().constData());
+    return 1;
+}
+
+/** ircGetConnectedHost()
+ *  Returns the Hostname of the connected IRC Server, as provided by the server itself.
+ *  This will return an empty string until the IRC Client has connected and received
+ *  a hostname message from the server.
+ */
+int TLuaInterpreter::ircGetConnectedHost(lua_State* L)
+{
+    Host* pHost = &getHostFromLua(L);
+    QString cHostName = "";
+    if (mudlet::self()->mpIrcClientMap[pHost]) {
+        cHostName = mudlet::self()->mpIrcClientMap[pHost]->getConnectedHost();
+    }
+
+    lua_pushstring(L, cHostName.toUtf8().constData());
+    return 1;
+}
+
+/** ircSaveNick( Nick )
+ *  Updates IRC client nickname configuration value.
+ *  Does not apply changes to active client until ircRestart() is called.
+ */
+int TLuaInterpreter::ircSaveNick(lua_State* L)
+{
+    string nick;
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "ircSaveNick: bad argument #1 type (Nick as string expected, got %s!)", lua_typename(L, 1));
+        return lua_error(L);
+    } else {
+        nick = lua_tostring( L, 1 );
+        if (nick.empty()) {
+            lua_pushnil(L);
+            lua_pushfstring(L, "ircSaveNick: bad argument #1 value (Nick must not be empty!)");
+            return 2;
+        }
+    }
+
+    Host* pHost = &getHostFromLua(L);
+    QPair<bool, QString> result = dlgIRC::writeIrcNickName(pHost, QString::fromStdString(nick));
+    if (!result.first) {
+        lua_pushnil(L);
+        lua_pushfstring(L, "unable to save nick name, reason: %s", result.second.toUtf8().constData());
+        return 2;
+    }
+
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+/** ircSaveServer( Hostname, Port )
+ *  Updates IRC client connection configuration values with the given Hostname and Port values.
+ *  Port argument is optional and defaults to 6667.
+ *  Does not apply changes to active client until ircRestart() is called.
+ */
+int TLuaInterpreter::ircSaveServer(lua_State* L)
```

Comment:
I'm not sure about this one.  Change seems like it would or should also cause the active connection to change but the function does not, only writes to config files from Lua.  I used Save as a convention from the other functions which deal with to-disk processes, but I think that is a bit ambiguous too.   Maybe `ircConfigServer()` or  `ircConfigureServer()` and similar changes for the nick and channel functions?